Lib/traceback.py

```diff
@@ -696,6 +696,39 @@ def emit(self, text_gen, margin_char=None):
                 yield textwrap.indent(text, indent_str, lambda line: True)
 
 
+def _resolve_exc_type(exc_type):
+    ATTRS = ('__name__', '__module__', '__qualname__')
+    if exc_type is None:
+        return None
+    elif isinstance(exc_type, type):
+        assert all(getattr(exc_type, name, None) for name in ATTRS)
+        return exc_type
+    elif all(getattr(exc_type, name, None) for name in ATTRS):
+        return exc_type
+    else:
+        raise ValueError(f'unsupported exc_type {exc_type!r}')
+
+
+def _match_class(exc_type, *expected):
```

Comment:
nit: I'd pass expected as a tuple (single arg), same way that types are passed to isinstance, etc.